### PR TITLE
Add GitAddAll

### PR DESCRIPTION
**Pull request by wekempf**

### CODE REVIEW

File: test.cake [hunk 1]
```diff
@@ -257,6 +257,14 @@ Task("Git-Modify-Add")
     GitAdd(testInitalRepo, testModifyFiles);
 });
 
+Task("Git-Modify-AddAll")
+    .IsDependentOn("Modify-Test-Files")
```
Comment (devlead) [inline test.cake]: You should modify some more files, so you add something not added by Git-Modify-Add task.

Comment (devlead) [inline test.cake]: The current is objective is to exercise the aliases on each platform as their own native libraries(test.cake is executed on 3 operating systems and 2 different build systems), libgit2sharp just proxies and interops to native code. If you don't change any files before staging it'll threat it as no-op and won't call native stage files as there are no changes.

Doing each step as a Cake task Felt natural in a cake script, also gives timings, handles exceptions and uses the code in a similar way as you would in a Cake script.
test.cake certainly isn't done nor perfect, and could certainly be refactored into helper methods for each task. It was mainly a quickfix to find xplat issues / non working aliases which have been an issue in the past, i.e. certain versions of libgit2sharp haven't worked on make and so on.

Happily take contributions to any improvements, just haven't had the time to refactor the script and just asked those that contribute new aliases to if possible add a simple test that exercises the added aliases so we at least have something.

Comment (wekempf) [inline test.cake]: Didn't mean anything I said as criticism, so I hope it was taken that way. Thanks for the answers. I'll do a minimal bit of refactoring here to provide the change requested, and possibly contribute further changes along this vein in the future.

**Comment by devlead** [comment]
@wekempf thanks for your contribution 👍

**Pull request merged**